server/src/services/checkForUpdates.js

```diff
@@ -1,32 +1,40 @@
 /* eslint-disable no-console */
 const { exec } = require('child_process')
+const fs = require('fs')
 
-const { version: currentVersion } = require('../../../package.json')
-const Fetch = require('./Fetch')
+try {
+  exec('git branch --show-current', async (err, stdout) => {
+    const gitRef = fs.readFileSync('.gitref', 'utf8')
 
-exec('git branch --show-current', async (err, stdout) => {
-  try {
-    if (err || typeof stdout !== 'string') {
-      throw new Error('Unable to get current branch', err.message)
+    if (!gitRef && (err || typeof stdout !== 'string' || !stdout.trim())) {
+      throw new Error('Unable to get current branch', err)
     }
-    const { version } = await Fetch.json(`https://raw.githubusercontent.com/WatWowMap/ReactMap/${stdout.trim()}/package.json`)
+    const branch = typeof gitRef === 'string' && gitRef.trim()
+      ? gitRef.trim()
```

Comment:
for branch name you need to remove prefix. content of the `.gitref`-file: `refs/heads/addl-version-checking`